### DISCUSSION TITLE
Absolute timestamps of RTSP streams and recording playbacks

### DISCUSSION THREAD

**Discussion by ArrietaUnai** [bluenviron/mediamtx#3174] [Questions]
### Question

Hi! I would like to ask a question about timestamps of RTSP streams when using MediaMTX as a proxy. 

I am using MediaMTX as a recorder of the RTSP stream. I would like to know the absolute timestamp of every frame. My goal is to get the timestamps of certain frames that will then be used to query a playback video. 

Following the explanation from this [reply](https://github.com/bluenviron/mediamtx/issues/1300#issuecomment-1641859910) I have tried to get the RTCP Sender Reports and I’m using the RTP offset to get the absolute timestamp, but the frame that I get from the recordings with these timestamps do not match. They are 4 or 5 frames off. 

This could be due to a problem with my implementation. But, I would like to ask:

Is the time that is used for the recordings the same as the one used in the RTSP stream RTCP Sender Reports? Should this method be accurate to get the desired frames from the recordings? 

Thanks in advance!

**Comment by aler9** [comment]
> Is the time that is used for the recordings the same as the one used in the RTSP stream RTCP Sender Reports? 

<s>Hello, the original timestamp contained into RTCP sender reports of incoming sources is currently discarded and replaced by the server timestamp as written in #1300. In recordings and in outgoing RTCP sender reports, the server timestamp is used. Therefore,  if you read a proxied stream from the server, you will have access to the exact absolute timestamp that is used in recordings.

Furthermore, getting frames from recordings is is quite complicated due to the fact that most frames require previous frames in order to be decoded. The [Playback server](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#playback-recorded-streams) is able to handle that logic for you.</s>
